lib/trailblazer/generator/generate.rb

```diff
@@ -0,0 +1,48 @@
+module Trailblazer
+  # Trailblazer Gen
+  class Generator
+    module Generate
+      extend Trailblazer::Activity::Railway()
+
+      REGEXP_CONCEPT = /^[A-Z][A-Za-z]{1,}(::[A-Z][A-Za-z]{1,})?/
+      REGEXP_NAME = /^[A-Z][A-Za-z]*$/
+
+      module_function
+
+      def validate_concept(_ctx, concept:, **)
+        concept.match REGEXP_CONCEPT
+      end
+
+      def validate_class_name(_ctx, options:, type:, **)
+        return Trailblazer::Activity::Right if type == :concept
+        options[:name].match REGEXP_NAME
+      end
+
+      def init(ctx, options:, concept:, type:, **)
+        concept = Utils::Fetch.concept(concept) unless type == :concept
+        ctx[:context] = Utils::Fetch.context(options, type, concept)
+      end
+
+      def generate_concept?(_ctx, type:, **)
+        type == :concept
+      end
+
+      def generate_concept(ctx, context:, **)
+        ctx[:destination] = destination = Concept.dir(context.concept_path)
+        Utils::Files.mkdir(destination)
+      end
+
+      # NOTE: why do I have to specifcy Trailblazer::Activity::End here and not in GenerateFile activity??
+      step method(:validate_concept), Output(:failure) => Trailblazer::Activity::End(:wrong_concept_format)
+      step method(:validate_class_name), Output(:failure) => Trailblazer::Activity::End(:wrong_class_name_format)
+      pass method(:init)
+      step method(:generate_concept?), Output(:success) => :generate_concept_step, Output(:failure) => :generate_file
+      step task: GenerateFile, id: :generate_file,
+           GenerateFile.outputs[:success] => "End.success",
```

Comment:
Here is where I was getting it wrong...calling `Trailblazer::Activity::End(:id)` will create a new `End` even-though in this case a `:success` end was already present. The correct one is using the `Output(:output) => :id` API where for the `End` is something like `"End.success"` so in this way I just connect that output to the existing `End`